### ISSUE TITLE
Bug : pagination broken when using 1 as perPage

### ISSUE THREAD

**Issue by BadeelProject** [codeigniter4/CodeIgniter4#1628]
---
name: Pagination breaks when using 1 row per page and removes the last page if it contains ONE ROW !
about: when using 1 row perPage, the paginator breaks

---
**Describe the bug**
- using 1 row per page breaks the paginator, and removes the last page if contains 1 row.
- also, when `groupBy` is used, the last page removes away if it rows count is less than previous pages' ones (_I do not know if this issue is related to my code or is common because it does not removes when groupBy is not used in THIS CASE_)
error message : `Call to a member function getPageCount() on null `

**CodeIgniter 4 version**
tested on alpha 4

**Affected module(s)**
database paginator

**Context**
 - OS: win7 / centos7
 - Web server : Apache
 - PHP version : 7.2.2

**Comment by nowackipawel** [comment]
I think it could be your code. 
Because one of last changes in pagination module are mine I decided to test it.

So, here is my code for pagination links:
```

		$pager = service('pager')->makeLinks($data['count']['page'], $data['count']['per_page'], $data['count']['query'], 'default_full', self::admin_object_model_page_number);
```
when I changed per_page for 1 everything is working correctly...when I filtered results to get only one is also working as it should.

![screenshot_2018-12-28_15-37-11](https://user-images.githubusercontent.com/8414802/50518576-a16b4c00-0ab6-11e9-9854-6d8ccb205f10.png)
![screenshot_2018-12-28_15-32-05](https://user-images.githubusercontent.com/8414802/50518577-a16b4c00-0ab6-11e9-9c2d-53c47cf7ef2b.png)
